x/cdp/abci.go

```diff
@@ -36,6 +40,13 @@ func BeginBlocker(ctx sdk.Context, req abci.RequestBeginBlock, k keeper.Keeper)
 			panic(err)
 		}
 
+		if skipSyncronizeAndLiquidations {
+			ctx.Logger().Debug(fmt.Sprintf("skipping x/cdp SynchronizeInterestForRiskyCDPs and LiquidateCdps for %s", cp.Type))
+			return
```

Comment:
should this be `continue` instead of `return`? Currently it'll update the first `CollateralParam` every block but return before it updates the others.

Reply:
If changed to `continue` it'll run `RunSurplusAndDebtAuctions` every block. If you want to optimise further I believe this function doesn't need to be run every block.

Reply:
agreed, updated to `continue` as `RunSurplusAndDebtAuctions` doesn't incur a significant performance penalty but we can always adjust in the future if that changes